tests/test_nexus_wrapper.py

```diff
@@ -1,12 +1,36 @@
+import numpy as np
+import pytest
 from mock import Mock
 
 from nexus_constructor.nexus.nexus_wrapper import (
     NexusWrapper,
     append_nxs_extension,
     get_nx_class,
 )
+from nexus_constructor.pixel_data import PixelMapping, PixelGrid, CountDirection, Corner
+from nexus_constructor.pixel_data_to_nexus_utils import (
+    pixel_grid_x_offsets,
+    pixel_grid_y_offsets,
+    pixel_grid_z_offsets,
+    pixel_grid_detector_ids,
+)
 from tests.test_nexus_to_json import create_in_memory_file
 
+global file_count
+file_count = 0
+
+
+@pytest.fixture(scope="function")
+def nexus_file():
+    global file_count
+    file_count += 1
+    return create_in_memory_file("test_nw" + str(file_count - 1))
```

Comment:
Rather than accumulating many in-memory files and having to give them unique names you could just close the file at the end of each test. It can be achieved something like this (I didn't test the code below):
```python
@pytest.fixture(scope="function")
def nexus_file(request):
    test_file = create_in_memory_file("test_nw")
    def fin():
        test_file.close()
    request.addfinalizer(fin)
    return test_file
```